PopPUNK/network.py

```diff
@@ -317,7 +328,16 @@ def constructNetwork(rlist, qlist, assignments, within_label,
                                                                               self = self_comparison))):
             if assignment == within_label:
                 if weights is not None:
-                    dist = np.linalg.norm(weights[row_idx, :])
+                    if weights_type == 'euclidean':
+                        dist = np.linalg.norm(weights[row_idx, :])
+                    elif weights_type == 'core':
+                        dist = weights[row_idx, 0]
+                    elif weights_type == 'accessory':
+                        dist = weights[row_idx, 1]
+                    else:
+                        sys.stderr.write('Unable to parse network weight type ' + weights_type + '; '
+                         'options are core, accessory or euclidean')
+                        sys.exit(1)
```

Comment:
Put this check and error at the start of the function (and also have as only possible options in argparse ideally)

Reply:
Validation added in 1b3b5bc; no menu options available to user, all definitions internal to code.